src/transform/LeadIntroductionTransform.js

```diff
@@ -0,0 +1,90 @@
+import Polyfill from './Polyfill'
+
+/**
+ * Determine if paragraph is the one we are interested in.
+ * @param  {!HTMLParagraphElement}  paragraphElement
+ * @return {!boolean}
+ */
+const isParagraphEligible = paragraphElement => {
+  // Ignore 'coordinates' which are presently hidden. See enwiki 'Bolton Field' and 'Sharya Forest
+  // Museum Railway'. Not counting coordinates towards the eligible P min textContent length
+  // heuristic has dual effect of P's containing only coordinates being rejected, and P's containing
+  // coordinates but also other elements meeting the eligible P min textContent length being
+  // accepted.
+  const coordElement = paragraphElement.querySelector('[id="coordinates"]')
+  const coordTextLength = !coordElement ? 0 : coordElement.textContent.length
+
+  // Ensures the paragraph has at least a little text. Otherwise silly things like a empty P or P
+  // which only contains a BR tag will get pulled up. See enwiki 'Hawaii', 'United States',
+  // 'Academy (educational institution)', 'Lovászpatona'
+  const minEligibleTextLength = 50
+  const hasEnoughEligibleText =
+    paragraphElement.textContent.length - coordTextLength >= minEligibleTextLength
+  return hasEnoughEligibleText
+}
+
+/**
+ * Nodes we want to move up. This includes the `eligibleParagraph` and everything up to (but not
+ * including) the next paragraph.
+ * @param  {!HTMLParagraphElement} eligibleParagraph
+ * @return {!Array.<Node>} Array of text nodes, elements, etc...
+ */
+const extractLeadIntroductionNodes = eligibleParagraph => {
+  const introNodes = []
+  let node = eligibleParagraph
+  do {
+    introNodes.push(node)
+    node = node.nextSibling
+  } while (node && !(node.nodeType === 1 && node.tagName === 'P'))
```

Comment:
Not a blocker but I've found it nice to have node type constants defined in a library so we can refer to them by symbols in our code: https://github.com/wikimedia/mediawiki-services-mobileapps/blob/master/lib/nodeType.js

Reply:
I added a single const for now, but I made a note to consider including definitions the way you mentioned - I'll audit the rest of the pagelib to see if we're using these elsewhere.